project2-beat-that/script.js

```diff
@@ -1,4 +1,124 @@
+    gameMode = "playerOneOrder";
+  }
+  // in the next game mode, player 1 to choose the order of their dice numbers
+  else if (gameMode === "playerOneOrder") {
+    console.log(gameMode);
+    // input validation
+    if (Number(input) !== 1 && Number(input) !== 2) {
+      myOutputValue = `Player One, you rolled ${playerOneNumbers[0]} and ${playerOneNumbers[1]}. <br> Please enter a valid order choice - 1 or 2`;
+      return myOutputValue;
+    }
+    // input order choice of 1 or 2
+    if (Number(input) === 1) {
+      playerOneScore = Number(
+        String(playerOneNumbers[0]) + String(playerOneNumbers[1])
+      );
+      myOutputValue = `Player One, you have chosen your number as ${playerOneScore}. <br> Player Two can roll now.`;
+      // change the game mode to player 2's turn to roll
+      gameMode = "playerTwoRoll";
+    }
+    if (Number(input) === 2) {
+      playerOneScore = Number(
+        String(playerOneNumbers[1]) + String(playerOneNumbers[0])
+      );
+      myOutputValue = `Player One, you have chosen your number as ${playerOneScore}. <br> Player Two can roll now.`;
+      // change the game mode to player 2's turn to roll
+      gameMode = "playerTwoRoll";
+    }
+
+    // player 2 game turn to roll the dice
+  } else if (gameMode === "playerTwoRoll") {
+    console.log(gameMode);
+    var rollDice1 = rollDice();
+    var rollDice2 = rollDice();
+    // to keep record of the dice numbers
+    playerTwoNumbers[0] = rollDice1;
+    playerTwoNumbers[1] = rollDice2;
+    // output dice number rolled and instruction for next game step
+    myOutputValue = `Player Two has rolled ${playerTwoNumbers[0]} and ${playerTwoNumbers[1]}. <br> Please choose order 1 or 2.`;
+    // change to the next game mode
+    gameMode = "playerTwoOrder";
+  }
+
+  // in the next game mode, player 2 to choose the order of their dice numbers
+  else if (gameMode === "playerTwoOrder") {
+    console.log(gameMode);
+    // input validation
+    if (Number(input) !== 1 && Number(input) !== 2) {
+      myOutputValue = `Player Two, you rolled ${playerTwoNumbers[0]} and ${playerTwoNumbers[1]}. <br> Please enter a valid order choice - 1 or 2`;
+    }
+    // player 2 input order choice of 1 or 2
+    if (Number(input) === 1) {
+      playerTwoScore = Number(
+        String(playerTwoNumbers[0]) + String(playerTwoNumbers[1])
+      );
+      myOutputValue = `Player Two, you have chosen your number as ${playerTwoScore}.`;
+      // change the game mode to results
+      gameMode = "gameResults";
+    }
+    if (Number(input) === 2) {
+      playerTwoScore = Number(
+        String(playerTwoNumbers[1]) + String(playerTwoNumbers[0])
+      );
+      myOutputValue = `Player Two, you have chosen your number as ${playerTwoScore}.`;
+      gameMode = "gameResults";
+    }
+  }
+
+  // to call the game results function to find out the winner
+  else if (gameMode === "gameResults") {
+    var winner = gameResults();
+    return winner;
```

Comment:
Instead of creating a new variable here and early returning it, you can also just do `myOutputValue = gameResults()`. But good attempt on refactoring here! 💯